src/averageAges.js

```diff
@@ -1,63 +1,77 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const averageMenAge = averageAge(
+    people.filter((person) => person.sex === 'm')
+  );
+
+  const menOfCentury = people.filter(
+    (person) => person.sex === 'm' && Math.ceil(person.died / 100) === century
+  );
+
+  const averageMenAgeOfCentury = averageAge(menOfCentury);
+
+  return century ? averageMenAgeOfCentury : averageMenAge;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenAverageAgeWithoutChild = averageAge(
+    people.filter((person) => person.sex === 'f')
+  );
+  const mothersName = people.map((person) => person.mother);
+
+  const womenAverageAgeWithChildren = averageAge(
+    people.filter((person) => mothersName.includes(person.name))
+  );
+
+  return withChildren
+    ? womenAverageAgeWithChildren
+    : womenAverageAgeWithoutChild;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersArr = people.map((person) => person.mother);
+  const childsArr = people.map((person) => person.name);
+
+  const mothers = people.filter((person) => mothersArr.includes(person.name));
+  const childs = people.filter((person) => childsArr.includes(person.mother));
+
+  return averageAgeDiff(mothers, childs, onlyWithSon);
 }
 
+const averageAge = (obj) => {
```

Comment:
better to use meaningful names, not just `obj`

Reply:
done